.scripts/CreateNuget.ps1

```diff
@@ -0,0 +1,201 @@
+    New-Item $nuget_dir_build -ItemType Directory -Force | Out-Null
+
+    Write-Host "Copying files..."
+
+    # Copy relevant contents of <FStar_root>/bin to tools/bin
+    ForEach ($bin_file in $bin_files) {
+        $bin_file_path = Join-Path -Path $fstarbin_path -ChildPath $bin_file
+        Copy-Item -Path $bin_file_path -Destination $nuget_dir_tools_bin -Force
+    }
+
+    # Do not include ulibfs build artifacts in the ulib
+    $ulib_obj = Join-Path -Path $ulib_path -ChildPath "obj"
+    $ulib_bin = Join-Path -Path $ulib_path -ChildPath "bin"
+    Remove-Item -Path $ulib_obj -Recurse -Force -ErrorAction Continue
+    Remove-Item -Path $ulib_bin -Recurse -Force -ErrorAction Continue
+        
+    # Copy <FStar_root>/ulib to tools/ulib
+    Copy-Item -Path $ulib_path -Destination "$nuget_dir_tools\ulib" -Recurse -Force
+        
+    # Copy <FStar_root>/bin/ulib.[dll|pdb|xml] to lib
+    Copy-Item -Path $ulibfs_path -Destination $nuget_dir_lib -Force
+
+    # Copy <FStar_root>/fsharp.extraction.targets to targets and rename it to $nuget_name.targets
+    Copy-Item -Path "$FStarPath\fsharp.extraction.targets" -Destination "$nuget_dir_build\$nuget_name.targets" -Force
+    Copy-Item -Path "$FStarPath\.nuget\$nuget_name.props" -Destination $nuget_dir_build -Force
+    # copy nuspec file and set version
+    Copy-Item -Path "$FStarPath\.nuget\$nuget_name.nuspec" -Destination $outputDir -Force
+
+    # Copy license files
+    # NOTE: Renaming LICENSE to LICENSE.txt as otherwise nuget is complaining about 'Part URI is empty'
+    Copy-Item -Path "$FStarPath\LICENSE" -Destination "$nuget_dir\LICENSE.txt" -Force
+    Copy-Item -Path "$FStarPath\LICENSE-fsharp.txt" -Destination "$nuget_dir\LICENSE-fsharp.txt" -Force
+    Copy-Item -Path "$FStarPath\LICENSE-z3.txt" -Destination "$nuget_dir\LICENSE-z3.txt" -Force
+
+    # Bump version in the copy of nuspec 
+    UpdateVersionInNuspec "$outputDir\$nuget_name.nuspec" $version
+
+    Write-Host "Packaging..."
+    & $NugetPath @("pack", "$outputDir\$nuget_name.nuspec", "-OutputDirectory", $outputDir)
+
+    Write-Host "Finished creating package"
+}
+
+function PushToNugetOrg{
+    [CmdletBinding()] param([string] $NugetExe, [string] $ApiKey, [string] $FStarNugetPath)
+    & $NugetExe @("push", $FStarNugetPath, "-ApiKey", $ApiKey)
+}
+
+$nuget_name = "FStar"
+$repo = "FStarLang/FStar"
+
+New-Item $outputDir -ItemType Directory -Force | Out-Null
+
+$package = FindBinaryPackage -repo $repo
+Write-Host "Selected package: $package"
+$FStarRootPath = GetWindowsBinaryPackage -package $package -outputDir $outputDir
+
+CompileUlibfs -MSBuildExe $msbuild -FStarRootPath $FStarRootPath
+
+PackageAsNuget -NugetPath $nuget -FStarPath $FStarRootPath -outputDir $outputDir -version $package.version
+
+$nuget_path = Join-Path -Path $outputDir -ChildPath "$nuget_name.$($package.version).nupkg"
+PushToNugetOrg -NugetExe $nuget -ApiKey $ApiKey -FStarNugetPath $nuget_path
```

Comment:
I've tested this up to actually publishing a nuget. I'll try to upload it as a dummy package with a changed name and on my account once I have the thumbs up from you.